src/pathOrderOptimizer.cpp

```diff
@@ -249,24 +321,74 @@ void LineOrderOptimizer::optimize()
             logError("Failed to find next closest line.\n");
         }
     }
+    if (loc_to_line != nullptr)
+    {
+        delete loc_to_line;
+    }
 }
 
-inline void LineOrderOptimizer::updateBestLine(unsigned int poly_idx, int& best, float& best_score, Point prev_point, Point incoming_perpundicular_normal)
+float LineOrderOptimizer::combingDistance2(const Point &p0, const Point &p1)
 {
+    if (loc_to_line == nullptr)
+    {
+        // do the initialisation required to be able to calculate realistic travel distances to the start of new paths
+        loc_to_line = PolygonUtils::createLocToLineGrid(*combing_boundary, 2000); // coarse 2mm grid to reduce computation time
+    }
+
+    CombPath comb_path;
+    if (LinePolygonsCrossings::comb(*combing_boundary, *loc_to_line, p0, p1, comb_path, -40, 0, false))
+    {
+        float dist = 0;
+        Point last_point = p0;
+        for (const Point& comb_point : comb_path)
+        {
+            dist += vSize(comb_point - last_point);
+            last_point = comb_point;
+        }
+        return dist * dist;
+    }
+
+    // couldn't comb, fall back to a large distance
+
+    return vSize2f(p1 - p0) * 10000;
+}
+
+inline void LineOrderOptimizer::updateBestLine(unsigned int poly_idx, int& best, float& best_score, Point prev_point, Point incoming_perpundicular_normal, int just_point)
+{
+    // when looking for chains, just_point will be either 0 or 1 depending on which vertex we are currently interested in testing
+    // if just_point is -1, it means that we are not looking for chains and we will test both vertices to see if either is best
+
     const Point& p0 = (*polygons[poly_idx])[0];
     const Point& p1 = (*polygons[poly_idx])[1];
-    float dot_score = getAngleScore(incoming_perpundicular_normal, p0, p1);
+    float dot_score = (just_point >= 0) ? 0 : getAngleScore(incoming_perpundicular_normal, p0, p1);
+
+    if (just_point != 1)
```

Comment:
This part can be slightly simplified (in a loop for example) so we don't need the duplicate the code, but I'm not sure how good its readability will be.

Reply:
Hi, sorry, I am not exactly sure to which part of the code you are referring to there. If you want something changed, please rephrase your requirement, thanks.